### PR TITLE
os/mac/xcode: support Xcode 12.5

### PR DESCRIPTION
**Pull request by Bo98**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

Also adjust Clang version handling to better differentiate patch versions. (Clang in Xcode 12.5 is Clang 12.0.5.)

### CODE REVIEW

File: Library/Homebrew/os/mac/xcode.rb [hunk 1]
```diff
@@ -201,25 +201,32 @@ def detect_version
...
+        version = DevelopmentTools.clang_version
+
+        return "dunno" if version.null?
 
         # This logic provides a fake Xcode version based on the
         # installed CLT version. This is useful as they are packaged
         # simultaneously so workarounds need to apply to both based on their
         # comparable version.
-        case (DevelopmentTools.clang_version.to_f * 10).to_i
+        version_integer = version.major.to_i * 100 + version.minor.to_i * 10 + version.patch.to_i
```
Comment (carlocab) [inline Library/Homebrew/os/mac/xcode.rb:212]: Is there any danger of double-digit minor/patch versions?

Comment (Bo98) [reply Library/Homebrew/os/mac/xcode.rb:212]: `DevelopmentTools.clang_version` does not currently support parsing double-digit minor/patch versions.

It's never happened before in the history of LLVM and Xcode.

Comment (carlocab) [reply Library/Homebrew/os/mac/xcode.rb:212]: Ok. I was just wondering why not just `(version * 100).to_i` here.

**Review by carlocab — DISMISSED** [review]

File: Library/Homebrew/os/mac/xcode.rb (hunk 1 above)
Comment (Bo98) [reply Library/Homebrew/os/mac/xcode.rb:212]: `to_f` strips the patch version.

Comment (Bo98) [reply Library/Homebrew/os/mac/xcode.rb:212]: To be honest, this could just be string comparisons.

**Comment by MikeMcQuaid** [comment]
Thanks @Bo98!

**Pull request merged**